rollup/internal/controller/permissionless_batches/minimal_recovery.go

```diff
@@ -0,0 +1,466 @@
+		return nil, nil, fmt.Errorf("last batch hash mismatch for batch %d, expected: %s, got: %s", event.BatchIndex(), args.LastBatchHash.String(), targetBatch.Hash().String())
+	}
+
+	// TODO: add support for multiple blob clients
+	blobClient := blob_client.NewBlobClients()
+	if r.cfg.RecoveryConfig.L1BeaconNodeEndpoint != "" {
+		client, err := blob_client.NewBeaconNodeClient(r.cfg.RecoveryConfig.L1BeaconNodeEndpoint)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create beacon node client: %w", err)
+		}
+		blobClient.AddBlobClient(client)
+	}
+
+	log.Info("Fetching blob by versioned hash and block time", "TargetBlobVersionedHash", targetBlobVersionedHash, "BlockTime", blockHeader.Time, "BlockNumber", blockHeader.Number)
+	blob, err := blobClient.GetBlobByVersionedHashAndBlockTime(r.ctx, targetBlobVersionedHash, blockHeader.Time)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get blob by versioned hash and block time for batch %d: %w", event.BatchIndex(), err)
+	}
+
+	daBlobPayload, err := codec.DecodeBlob(blob)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode blob for batch %d: %w", event.BatchIndex(), err)
+	}
+
+	return targetBatch, daBlobPayload, nil
+}
+
+func (r *MinimalRecovery) fetchL2Blocks(fromBlock uint64, l2BlockHeightLimit uint64) (uint64, error) {
+	if l2BlockHeightLimit > 0 && fromBlock > l2BlockHeightLimit {
+		return 0, fmt.Errorf("fromBlock (latest finalized L2 block) is higher than specified L2BlockHeightLimit: %d > %d", fromBlock, l2BlockHeightLimit)
+	}
+
+	log.Info("Fetching L2 blocks with", "fromBlock", fromBlock, "l2BlockHeightLimit", l2BlockHeightLimit)
+
+	// Fetch and insert the missing blocks from the last block in the batch to the latest L2 block.
+	latestL2Block, err := r.l2Watcher.Client.BlockNumber(r.ctx)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get latest L2 block number: %w", err)
+	}
+
+	log.Info("Latest L2 block number", "latest L2 block", latestL2Block)
+
+	if l2BlockHeightLimit > latestL2Block {
+		return 0, fmt.Errorf("l2BlockHeightLimit is higher than the latest L2 block number, not all blocks are available in L2geth: %d > %d", l2BlockHeightLimit, latestL2Block)
+	}
+
+	toBlock := latestL2Block
+	if l2BlockHeightLimit > 0 {
+		toBlock = l2BlockHeightLimit
+	}
+
+	err = r.l2Watcher.GetAndStoreBlocks(r.ctx, fromBlock, toBlock)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get and store blocks: %w", err)
+	}
+
+	log.Info("Fetched L2 blocks from", "fromBlock", fromBlock, "toBlock", toBlock)
+
+	return toBlock, nil
+}
+
+func (r *MinimalRecovery) resetDB() error {
+	sqlDB, err := r.db.DB()
+	if err != nil {
+		return fmt.Errorf("failed to get db connection: %w", err)
+	}
+
+	// reset and init DB
+	var v int64
+	err = migrate.Rollback(sqlDB, &v)
+	if err != nil {
+		return fmt.Errorf("failed to rollback db: %w", err)
+	}
+
+	err = migrate.Migrate(sqlDB)
+	if err != nil {
+		return fmt.Errorf("failed to migrate db: %w", err)
+	}
```

Comment:
```suggestion
	// reset and init DB
	var v int64
	err = migrate.Rollback(sqlDB, &v)
	if err != nil {
		return fmt.Errorf("failed to rollback db: %w", err)
	}

	err = migrate.Migrate(sqlDB)
	if err != nil {
		return fmt.Errorf("failed to migrate db: %w", err)
	}
```

This (rollback to `0` version then migrate to the latest version) can be simplified by `migrate.Reset(sqlDB)`.

Reply:
Done in 4f884028cca670ead4d92b768f7ecb673f148dda